tools/merge_requirements.py

```diff
@@ -1,59 +1,74 @@
-    :rtype: dict
 
+def read_requirement_file(path, data):
     """
-    d = {}
-    with open(file_path) as f:
-        for line in f:
-            line = line.strip()
-            if line and not line.startswith('#'):
-                project, version = line.split('==')
-                if not version:
-                    raise ValueError("Unexpected syntax '{0}'".format(line))
-                d[project] = version
-    return d
-
+    Read a requirement file, and feeds its content into the given data dict.
+    :param str path: the requirement file path to read
+    :param dict data: the data dict to feed with
+    """
+    with open(path) as fh:
+        for line in fh:
+            if line.strip() and not line.startswith('#') and not line.startswith('-e'):
+                match = REQUIREMENT_REGEX.match(line)
 
-def output_requirements(requirements):
-    """Prepare print requirements to stdout.
+                if not match:
+                    raise ValueError("Unexpected syntax '{0}'".format(line))
 
-    :param dict requirements: mapping from a project to its pinned version
+                package = match.group(1)
+                version = LooseVersion(match.group(2))
 
-    """
-    return '\n'.join('{0}=={1}'.format(k, v)
-          for k, v in sorted(requirements.items()))
+                if not data.get(package):
+                    data[package] = []
 
+                data[package].append(version)
 
-def main(*files):
-    """Merges multiple requirements files together and prints the result.
 
-    Requirement files specified later in the list take precedence over earlier
-    files.
+def merge_requirements(data):
+    """
+    Merge requirements in the dict data, by returning one requirement for each package.
+    First version found for each package is retained.
+    :param dict data: dict of all requirements
+    :return: the merged requirements
+    :rtype: list
+    """
+    merged_data = []
+    for package, versions in data.items():
+        merged_data.append((package, str(versions[-1])))
+    return merged_data
 
-    :param tuple files: paths to requirements files
 
+def main(*paths):
+    """
+    Main function of this module.
+    Accept a list of requirements files, return a list of well formatted merged requirements.
+    Order of paths follows an increasing priority: version retained for a given package will be
+    the last one found along all requirements files.
+    :param str paths: list of the requirement files to merge
+    :return: a well formatted merged requirements
+    :rtype: str
     """
-    d = {}
-    for f in files:
-        d.update(read_file(f))
-    return output_requirements(d)
+    data = {}
+    for path in paths:
+        read_requirement_file(path, data)
+
+    requirements = merge_requirements(data)
+    requirements.sort(key=lambda requirement: requirement[0].lower())
+    return os.linesep.join(['=='.join(requirement) for requirement in requirements])
```

Comment:
I don't think we should use `linesep` here. Docs for [os.linesep](https://docs.python.org/3/library/os.html#os.linesep) say that it should not be used when writing to a text file and [print](https://docs.python.org/3/library/functions.html#print) writes to text files.

Reply:
Reverted to original implementation.